index.js

```diff
@@ -321,10 +321,23 @@ Assert.add('eitherOfType, oneOfType', function multitypecheck(ofs, msg) {
   var value = type(this.value)
     , expect = format('`%j` (%s) to @ be a %s', this.value, value, ofs.join(' or a '));
 
-  return this.test(ofs.some(function(of) {
-    of = of.toString().toLowerCase();
-    return of === value;
-  }), msg, expect);
+  var test;
+  if (typeof ofs.some === 'function') {
```

Comment:
There is no need for the ofs.some because you already have a normal unrolled version of it below. This just adds duplicate code just for the sake of using array.some. The only thing that needs to be here is the for loop. Just ditch array.some completely.